### ISSUE TITLE
BCM5720 support, BCM5720 H2B support, BCM5721(?) support

### ISSUE THREAD

**Issue by hlandau** [meklort/bcm5719-fw#272]
Various people have asked me about:

- BCM5720 support
- Support for the BCM5720 host-to-BMC feature
- BCM5721 support

No idea of @meklort's view but I just wanted to note here that I can provide documentation on at least the BCM5720-related aspects of the APE/MIPS firmware if needed, if someone (not necessarily @meklort) wanted to undertake the cleanroom side. I have some decodes of the H2B registers somewhere in particular.

BCM5721 is a consumer variant of the Tigon line, reportedly without the APE. It may not be too hard to support but have not investigated at this time.

**Comment by meklort** [comment]
I also received an email asking about support for the BCM5720 (and sent a reply), but the other big item related to support is that I don't have NC-SI over SMBus. That would be needed for add-in-card support.

For something like the BCM5721 w/o APE - more features would need to be supported here, specifically PXE boot by providing an option rom (I am interested in that though), so that something like gPXE/Etherboot could be provided. This complicates things when trying to provide a signed UEFI option rom. I had a quick discussion with @hughsie when originally getting this into LVFS that it wasn't a goal at the time, but It would be nice to enable for off the shelf cards.

So, features that are known to need additional changes in the firmware (and will require support from you @hlandau):
- H2B
- SMBus
- Option ROM support
- Jumbo Frames (#269)

**Comment by hlandau** [comment]
@meklort Yep. It shouldn't be hard for me to get the notes on how to use these features for you, since I retain my annotated disassembly. I will try post something for these features. Do you want to do H2B or SMBus or Option ROMs first (which should I prioritise)?

I will post the docs here or in the ortega repository, but of course, you can always ping me with questions in #t-w like we have done previously.

**Comment by meklort** [comment]
I'm personally more interested in the SMBus and H2B access items, so I'd suggest those.
